Source/Dafny/Dafny.atg

```diff
@@ -2625,9 +2625,9 @@ Rhs<out AssignmentRhs r>
        }
     .)
   | "*"                                (. r = new HavocRhs(t); .)
-  | Expression<out e, false, true>     (. r = new ExprRhs(e); .)
+  | Expression<out e, false, true>     (. r = e != null ? new ExprRhs(e) : null; .)
   )
-  { Attribute<ref attrs> }             (. r.Attributes = attrs; .)
+  { Attribute<ref attrs> }             (. if(r != null) { r.Attributes = attrs; } .)
```

Comment:
Is it possible that this can lead to some portion of the source code being silently ignored? And, if so, can we detect it and emit a useful error message?

Reply:
I don't know to what extent it will ignore portions of the source code, because before it would actually crash. The new behavior is thus strictly better than the previous one, but again I can't guarantee the absence of null pointer exceptions.